File: reader-test/src/component-test/java/com/richo/reader/test/pages/LoginPage.java
package com.richo.reader.test.pages;

import com.jayway.restassured.RestAssured;

import java.util.Optional;

public class LoginPage
{
	private final String baseUrl;
	private Optional<String> token;
	private Optional<String> username;

	public LoginPage(final int port)
	{
		baseUrl = "http://localhost:" + port;
	}

	public void createUser(final String username)
	{
		RestAssured
				.given().body(username)
				.when().post(baseUrl + "/api/users")
				.then().assertThat().statusCode(200);
	}

	public void login(final String username)
	{
		login(username, "123456789qwertyuio123qweasd");
	}

	public void login(String username, String password)
	{
		this.username = Optional.of(username);
		this.token = Optional.ofNullable(RestAssured
				.given().body(password)
				.when().post(baseUrl + "/api/users/" + username + "/sessions")
				.then().extract().body().jsonPath().get("token"));
	}

	public boolean isLoggedIn()
	{
		return token.isPresent();
	}

	public FeedPage toFeedPage()
	{
		return new FeedPage(
				baseUrl,
				username.orElseThrow(() -> new RuntimeException("No username")),
				token.orElseThrow(() -> new RuntimeException("No token")));
	}
}

File: autoplaylist-backend/server/spring/src/test/kotlin/com/richodemus/autoplaylist/test/pages/LoginPage.kt
package com.richodemus.autoplaylist.test.pages

import io.restassured.RestAssured
import io.restassured.http.ContentType.JSON

class LoginPage(private val port: Int) {
    fun login(code: String): MainPage {
        val sessionId = RestAssured
                .given().body("""{"code":"$code"}""").contentType(JSON)
                .`when`().post("http://localhost:$port/v1/sessions")
                .then().assertThat().statusCode(200).extract().detailedCookie("JSESSIONID")
        return MainPage(port, sessionId)
    }
}
